modules/expression-compiler/src/main/scala-3/dotty/tools/dotc/evaluation/ExtractExpression.scala

```diff
@@ -68,7 +68,12 @@ class ExtractExpression(using exprCtx: ExpressionContext) extends MacroTransform
       desugaredIdent match
         case tree: ImportOrExport => tree
 
-        case tree if tree.symbol.is(Inline) => tree
+        case tree if tree.symbol.is(Inline) =>
+          tree.symbol.info match
+            case ConstantType(value) => Literal(value)
```

Comment:
It is safer to use `cpy(tree).Literal(value)` to not lose all the attachements that we can have on `tree`.

```suggestion
            case ConstantType(value) => cpy(tree).Literal(value)
```